pandas/core/arrays/integer.py

```diff
@@ -196,7 +196,11 @@ def coerce_to_array(values, dtype, mask=None, copy=False):
             mask = mask.copy()
         return values, mask
 
-    values = np.array(values, copy=copy)
+    if hasattr(values, "dtype"):
+        values = np.array(values, copy=copy)
```

Comment:
What cases actually hit this now? Wouldn't this still lose precision?

Reply:
instances like pd.Series, Index, and np.array hits this. 
And it will only lose precision when the argument `values` is something with `dtype=float`(which is not wrong I think). Because it will get the values that are already rounded.